core/src/main/java/org/apache/calcite/jdbc/CalciteSchema.java

```diff
@@ -103,6 +105,11 @@ protected CalciteSchema(CalciteSchema parent, Schema schema,
       this.functionNames = Preconditions.checkNotNull(functionNames);
       this.nullaryFunctionMap = Preconditions.checkNotNull(nullaryFunctionMap);
     }
+    if (typeMap == null) {
+      this.typeMap = new NameMap<>();
+    } else {
+      this.typeMap = Preconditions.checkNotNull(typeMap);
```

Comment:
is this null check necessary ?

Reply:
the initializers for the other fields do something similar. it helps the IDE deduce that the field is never null.